### PR TITLE
PlayJson: Use a RejectionError to transport a parsing error

### PR DESCRIPTION
**Pull request by lenniboy**
As discussed in #46, akka-http allows you to put more information about unmarshalling errors into the rejections.

This PR puts the original `JsError` into the `Rejection`. Unfortunately I had to create a pretty superfluous wrapper exception but I haven't found a way to avoid this.

### CODE REVIEW

**Comment by lenniboy** [comment]
Also, it took me a little while to figure out the actual rejection handler code, which is this one:

```scala
implicit def rejectionHandler =
    RejectionHandler
      .newBuilder()
      .handleAll[ValidationRejection] { rejections =>
        val combinedError =
          rejections.collect({ case ValidationRejection(_, Some(PlayJsonError(jsError))) => jsError }).reduce(_ ++ _)

        val errorJson = Json.stringify(JsError.toJson(combinedError))
        complete((StatusCodes.BadRequest, HttpEntity(ContentTypes.`application/json`, errorJson)))
      }
      .result()
```

This should be documented somewhere. Where would be a good place? In the readme, or maybe create one in `akka-http-play-json/Readme.md`?

File: akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala [hunk 1]
```diff
@@ -78,5 +81,11 @@ trait PlayJsonSupport {
       implicit writes: Writes[A],
       printer: JsValue => String = Json.prettyPrint
   ): ToEntityMarshaller[A] =
-    jsonStringMarshaller.compose(printer).compose(writes.writes)
+    Marshaller.StringMarshaller
```
Comment (hseeberger) [inline akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala]: What's the reason for this change?

File: akka-http-play-json/src/test/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupportSpec.scala [hunk 2]
```diff
@@ -19,12 +19,14 @@ package de.heikoseeberger.akkahttpplayjson
 import akka.actor.ActorSystem
 import akka.http.scaladsl.marshalling.Marshal
 import akka.http.scaladsl.model.ContentTypes.`application/json`
-import akka.http.scaladsl.model.{ HttpEntity, MediaTypes, RequestEntity }
+import akka.http.scaladsl.model.{HttpEntity, MediaTypes, RequestEntity}
```
Comment (hseeberger) [inline akka-http-play-json/src/test/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupportSpec.scala]: Please `test:compile` with sbt to format the code.

File: akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala (hunk 1 above)
Comment (lenniboy) [inline akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala]: Oh, that must haven been a copy & paste error.

**Comment by lenniboy** [comment]
I've fixed what came up in the review and amended my commit. It should be all clean now.

File: akka-http-play-json/src/test/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupportSpec.scala (hunk 2 above)
Comment (lenniboy) [inline akka-http-play-json/src/test/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupportSpec.scala]: Also fixed this.

**Comment by hseeberger** [comment]
Thanks!
